### PR TITLE
fixed validate cert settings issue

### PR DESCRIPTION
**Pull request by manurodriguez**
##### SUMMARY

Normal update mandatory parameters for container certification project failed to validate-cert-setting

##### ISSUE TYPE

Fixes:  https://github.com/redhatci/ansible-collection-redhatci-ocp/issues/482

### CODE REVIEW

File: roles/create_certification_project/tasks/validate_cert_settings.yml [hunk 1]
```diff
@@ -15,7 +15,30 @@
...
+    - cert_settings.build_categories
+    - cert_settings.registry_override_instruct
+    - cert_settings.application_categories
+    - cert_settings.os_content_type
+    - cert_settings.privileged
+    - cert_settings.release_category
+    - cert_settings.repository_description
+  when:
+    - product_type == 'container'
+    - do_container_parallel_test == false
```
Comment (manurodriguez) [inline roles/create_certification_project/tasks/validate_cert_settings.yml:41]: I was discussing with Andrew about the lint error, we believe the problem is here, and it should be changed for something like:
```suggestion
    - do_container_parallel_test is not True
```
or even better:
```suggestion
    - not do_container_parallel_test | default(False) | bool
```

File: roles/create_certification_project/tasks/validate_cert_settings.yml [hunk 2]
```diff
@@ -15,7 +15,30 @@
       - "cert_settings.repository_description is defined"
     fail_msg: "The parameter {{ item }} is required to update cert project"
     success_msg: "The parameter {{ item }} is present"
-  when: product_type == 'container'
+  when:
+    - product_type == 'container'
+    - do_container_parallel_test
```
Comment (manurodriguez) [inline roles/create_certification_project/tasks/validate_cert_settings.yml:20]: Can we add a condition to this line to make it clear? something like:
```suggestion
    - do_container_parallel_test | default(False) | bool
```

[the remark this answers is not in the document]
Comment (ansvu) [reply roles/create_certification_project/tasks/validate_cert_settings.yml]: Thanks, Tony!

File: roles/create_certification_project/tasks/validate_cert_settings.yml (hunk 1 above)
Comment (ansvu) [reply roles/create_certification_project/tasks/validate_cert_settings.yml]: Thanks, Manny!

File: roles/create_certification_project/tasks/validate_cert_settings.yml [hunk 3]
```diff
@@ -15,7 +15,30 @@
...
     fail_msg: "The parameter {{ item }} is required to update cert project"
     success_msg: "The parameter {{ item }} is present"
-  when: product_type == 'container'
+  when:
+    - product_type == 'container'
+    - do_container_parallel_test | bool
+
+- name: Validate cert_settings for container project
+  ansible.builtin.assert:
+    that: "{{ item }} is defined"
```
Comment (tkrishtop) [inline roles/create_certification_project/tasks/validate_cert_settings.yml:24]: I believe it's better to assign a custom variable to the `item` in order to validate the assertion

File: roles/create_certification_project/tasks/validate_cert_settings.yml [hunk 4]
```diff
@@ -15,7 +15,30 @@
...
+    - cert_settings.build_categories
+    - cert_settings.registry_override_instruct
+    - cert_settings.application_categories
+    - cert_settings.os_content_type
+    - cert_settings.privileged
+    - cert_settings.release_category
+    - cert_settings.repository_description
+  when:
+    - product_type == 'container'
+    - not do_container_parallel_test | bool
```
Comment (tkrishtop) [inline roles/create_certification_project/tasks/validate_cert_settings.yml:41]: I'm against adding "if parallelization / if not parallelization" everywhere, to be honest.
Will review the code if we could structure this in a more generic way

Comment (wying3) [reply roles/create_certification_project/tasks/validate_cert_settings.yml:41]: are you checking this and  proposal new solution?

**Comment by ansvu** [comment]
recheck

**Comment by ansvu** [comment]
recheck

**Comment by ansvu** [comment]
recheck

**Comment by ansvu** [comment]
recheck

**Comment by ansvu** [comment]
recheck

File: roles/create_certification_project/tasks/validate_cert_settings.yml (hunk 3 above)
Comment (manurodriguez) [reply roles/create_certification_project/tasks/validate_cert_settings.yml:24]: maybe something like this: `loop_var: container_setting`?
```
- name: Validate cert_settings for container project
  ansible.builtin.assert:
    that: "{{ container_setting }} is defined"
    fail_msg: "The parameter {{ container_setting }} is required to update cert project"
    success_msg: "The parameter {{ container_setting }} is present"
  loop:
    - cert_item.short_description
    - cert_item.short_description | length >= 50
    - cert_settings.auto_publish
    - cert_settings.email_address
    - cert_settings.build_categories
    - cert_settings.registry_override_instruct
    - cert_settings.application_categories
    - cert_settings.os_content_type
    - cert_settings.privileged
    - cert_settings.release_category
    - cert_settings.repository_description
  loop_control:
    loop_var: container_setting
  when:
    - product_type == 'container'
    - not do_container_parallel_test | bool
```

File: roles/create_certification_project/tasks/validate_cert_settings.yml (hunk 4 above)
Comment (tkrishtop) [reply roles/create_certification_project/tasks/validate_cert_settings.yml:41]: Hi @wying3 I'll fix and re-test the PR seeing its urgency. I hope next time @ansvu could find some time to check the code reviews :(

File: roles/create_certification_project/tasks/validate_cert_settings.yml [hunk 5]
```diff
@@ -15,7 +15,30 @@
...
+    - do_container_parallel_test
+
+- name: Validate cert_settings for container project
+  ansible.builtin.assert:
+    that: "{{ item }} is defined"
+    fail_msg: "The parameter {{ item }} is required to update cert project"
+    success_msg: "The parameter {{ item }} is present"
+  with_items:
+    - cert_item.short_description
+    - cert_item.short_description | length >= 50
```
[the remark this answers is not in the document]
Comment (manurodriguez) [reply roles/create_certification_project/tasks/validate_cert_settings.yml]: This was addressed in the last commit 775294b91fb6bcbf551db53c33bdaceb7a29c45f

File: roles/create_certification_project/tasks/validate_cert_settings.yml (hunk 3 above)
Comment (manurodriguez) [reply roles/create_certification_project/tasks/validate_cert_settings.yml:4]: we somehow replaced the `item` var, not that in play exactly, please see commit: 775294b91fb6bcbf551db53c33bdaceb7a29c45f

**Comment by tkrishtop** [comment]
LGTM, thank you for all the changes @ansvu and @manurodriguez !

**Review by tkrishtop — APPROVED** [review]

**Comment by tkrishtop** [comment]
##### Tests

- [x]  normal-container-validation - https://www.distributed-ci.io/jobs/aed8aae2-36cf-40c6-aa69-9ef7d4bb4326/jobStates?sort=date - (Nov 13)
- [x]  parallelization-container-validation - https://www.distributed-ci.io/jobs/3f973e4a-775c-4acf-a312-3f3bc2eb3e1c/jobStates?sort=date - (Nov 13)
- [x]  preflight-green - https://www.distributed-ci.io/jobs/843b1df4-6c15-4718-8378-21e6a5e7298b/jobStates?sort=date -(Nov 13)

build-depends: https://github.com/dci-labs/bos2-ci-config/pull/219

TestBos2Sno: sno normal-container-validation parallelization-container-validation

**Pull request merged**